src/plugins/intel_cpu/src/nodes/strided_slice.cpp

```diff
@@ -38,8 +40,110 @@ bool StridedSlice::isSupportedOperation(const std::shared_ptr<const ov::Node>& o
     return true;
 }
 
+class StridedSliceShapeInfer : public ShapeInferEmptyPads {
+public:
+    StridedSliceShapeInfer(const IShapeInfer::port_mask_t& port_mask,
+                           size_t output_size,
+                           std::unordered_set<int64_t> begin_mask,
+                           std::unordered_set<int64_t> end_mask,
+                           std::unordered_set<int64_t> new_axis_mask,
+                           std::unordered_set<int64_t> shrink_axis_mask)
+    : m_port_mask(port_mask), m_outputShape(output_size, 1),
+      m_begin_mask_set(std::move(begin_mask)),
+      m_end_mask_set(std::move(end_mask)),
+      m_new_axis_mask_set(std::move(new_axis_mask)),
+      m_shrink_axis_mask_set(std::move(shrink_axis_mask)) {}
+
+    Result infer(
+        const std::vector<std::reference_wrapper<const VectorDims>>& input_shapes,
+        const std::unordered_map<size_t, MemoryPtr>& data_dependency) override {
+        const VectorDims& shapeIn = input_shapes[0].get();
+        const VectorDims& shapeBegin = input_shapes[1].get();
+        if (data_dependency.at(1)->GetDataType() != dnnl::memory::data_type::s32) {
+            IE_THROW(Unexpected) << "The data type of begin/end/stride is NOT s32, which is unexpected!";
+        }
```

Comment:
Two comments:
1. It is better to use `data_dependency.at(1)->getDesc().getPrecision() != Precision::I32` to avoid extra dependency on dnnl entities.
2. It looks like we need additional checks for the other 2 componentis, don't we?

Reply:
> Two comments:
> 
> 1. It is better to use `data_dependency.at(1)->getDesc().getPrecision() != Precision::I32` to avoid extra dependency on dnnl entities.
> 2. It looks like we need additional checks for the other 2 componentis, don't we?

apply them.